server/src/main/java/org/cloudfoundry/identity/uaa/zone/MultitenantVmidentityClientDetailsService.java

```diff
@@ -333,4 +364,54 @@ private static ClientDetails mapOIDCClient(OIDCClient client, String tenant, Cas
         return list;
     }
 
+    private void sortClientDetails(List<ClientDetails> list, String sortBy, boolean ascending) {
```

Comment:
Each of these compare functions is different and relies on the underlying structure of the contents of the list - because it has to map the SCIM filter attribute (e.g. `"authorized_grant_types"` to the getter). The SCIM attributes don't really match up with their respective getters (e.g. `"web_server_redirect_uri"` => `#getRegisteredRedirectURI()`). 

Setting up some structure to support both the arbitrary mapping _and_ perform the the comparison winds up leading to the same sort of structure: a map of attribute names and then lambdas containing comparison functions.